Rework demo example

Lowentwickler opened the pull request.

ZKpot commented:
Everything seems to be working. 
Please fix the textures in skybox demo.
Also I would add more basic controls to the demo (ASD and jump), it really feels incomplete without them.

ZKpot commented:
9d5062c broke skybox demo:
```
Finished release [optimized] target(s) in 1m 21s
     Running `target\release\examples\skybox.exe`
Starting SkyBox Example
[0.795442 ERROR]()(no module): [0] Resource import from `"assets/skybox/skybox_right.png"` failed: 
        FileRead(Os { code: 2, kind: NotFound, message: "Не удается найти указанный файл." })
[0.802122 ERROR]()(no module): [2] Resource import from `"assets/skybox/skybox_front.png"` failed: 
        FileRead(Os { code: 2, kind: NotFound, message: "Не удается найти указанный файл." })
[0.802218 ERROR]()(no module): [3] Resource import from `"assets/skybox/skybox_back.png"` failed:
        FileRead(Os { code: 2, kind: NotFound, message: "Не удается найти указанный файл." })
```

Lowentwickler commented:
I've fixed the skybox example and also added more changes:
- Fix for lights, there was no transformation for model's normals
- Assets moved to appropriate examples folders
- Add more comments to demo
- Add overlay example
- Fix for skybox texture flipping
- Changed skybox and terrain texture for the Demo (brighter and has direction indicators)
